pilot/pkg/leaderelection/leaderelection.go

```diff
@@ -70,6 +71,7 @@ func (l *LeaderElection) Run(stop <-chan struct{}) {
 			// We were told to stop explicitly. Exit now
 			return
 		default:
+			cancel()
```

Comment:
which context is this? Is this the overall pilot run context? Or a new one created just for each component